loopback-db-migrate.js

```diff
@@ -2,194 +2,218 @@
+  },
+  "db": {
+    "type": "String",
+    "length": 100,
+    "required": true
+  },
+  "runDtTm": {
+    "type": "Date",
+    "required": true
+  }
 });
 
 // make migration folders if they don't exist
 try {
-    fs.mkdirSync(migrationsFolder);
+  fs.mkdirSync(migrationsFolder);
 } catch (e) {}
 try {
-    fs.mkdirSync(dbMigrationsFolder);
+  fs.mkdirSync(dbMigrationsFolder);
 } catch (e) {}
 
 function mapScriptObjName(scriptObj){
-    return scriptObj.name;
+  return scriptObj.name;
 }
 
 function findScriptsToRun(upOrDown, cb) {
-    var filters = {
-        where: {
-            name: { gte: dateSinceFilter+'' || '' }
-        },
-        order: (upOrDown === 'up' ) ? 'name ASC' : 'name DESC'
-    };
-
-    // get all local scripts and filter for only .js files
-    var localScriptNames = fs.readdirSync(dbMigrationsFolder).filter(function(fileName) {
-        return fileName.substring(fileName.length - 3, fileName.length) === '.js';
+  var filters = {
+    where: {
+      name: { gte: dateSinceFilter+'' || '' }
+    },
+    order: (upOrDown === 'up' ) ? 'name ASC' : 'name DESC'
+  };
+
+  // get all local scripts and filter for only .js files
+  var localScriptNames = fs.readdirSync(dbMigrationsFolder).filter(function(fileName) {
+    return fileName.substring(fileName.length - 3, fileName.length) === '.js';
+  });
+
+  // create table if not exists
+  datasource.autoupdate('Migration', function (err) {
+    if (err) {
+      console.log('Error retrieving migrations:');
+      console.log(err.stack);
+      process.exit(1);
+    }
+
+    // get all scripts that have been run from DB
+    datasource.models.Migration.find(filters, function (err, scriptsRun) {
+      if (err) {
+        console.log('Error retrieving migrations:');
+        console.log(err.stack);
+        process.exit(1);
+      }
+
+      if (upOrDown === 'up') {
+        var runScriptsNames = scriptsRun.map(mapScriptObjName);
+
+        // return scripts that exist on disk but not in the db
+        cb(localScriptNames.filter(function (scriptName) {
+          return runScriptsNames.indexOf(scriptName) < 0;
+        }));
+      } else {
+        // return all db script names
+        cb(scriptsRun.map(mapScriptObjName));
+      }
     });
+  });
+}
 
-    // create table if not exists
-    datasource.autoupdate('Migration', function (err) {
-        if (err) {
-            console.log('Error retrieving migrations:');
-            console.log(err.stack);
-            process.exit(1);
-        }
+function waitForAppToBeBooted(done) {
+  const INTERVAL = 50;
+  const TIMEOUT = 1000;
+  let internalId;
+  let count = 0;
 
-        // get all scripts that have been run from DB
-        datasource.models.Migration.find(filters, function (err, scriptsRun) {
-            if (err) {
-                console.log('Error retrieving migrations:');
-                console.log(err.stack);
-                process.exit(1);
-            }
+  internalId = setInterval(async () => {
```

Comment:
Not sure we get anything from this being async.

Reply:
You are totally right.